ext/bcmath/bcmath.c

```diff
@@ -777,4 +791,965 @@ PHP_FUNCTION(bcscale)
+		goto fallback;
+	}
+
+	if (UNEXPECTED(Z_TYPE_P(op2) == IS_NULL)) {
+		lval2 = 0;
+	} else if (UNEXPECTED(bcmath_number_parse_num(op2, &obj2, &str2, &lval2) == FAILURE)) {
+		goto fallback;
+	}
+
+	if (UNEXPECTED(bc_num_from_obj_or_str_or_long(&n1, NULL, obj1, str1, lval1) == FAILURE ||
+		bc_num_from_obj_or_str_or_long(&n2, NULL, obj2, str2, lval2) == FAILURE)) {
+		goto fallback;
+	}
+
+	int ret = bc_compare(n1, n2, MAX(n1->n_scale, n2->n_scale));
+
+	if (Z_TYPE_P(op1) != IS_OBJECT) {
+		bc_free_num(&n1);
+	}
+	if (Z_TYPE_P(op2) != IS_OBJECT) {
+		bc_free_num(&n2);
+	}
+
+	return ret;
+
+fallback:
+	return zend_std_compare_objects(op1, op2);
+}
+
+#define BCMATH_PARSE_PARAMETERS_START(min_num_args, max_num_args) \
+	bool bc_num_arg_error = false; \
+	ZEND_PARSE_PARAMETERS_START(min_num_args, max_num_args)
+
+#define BCMATH_PARAM_NUMBER_OR_STR_OR_LONG(dest_obj, ce, dest_str, dest_long) \
+	Z_PARAM_PROLOGUE(0, 0); \
+	if (UNEXPECTED(!(zend_parse_arg_obj(_arg, &dest_obj, ce, 0) || \
+		zend_parse_arg_str_or_long(_arg, &dest_str, &dest_long, &_dummy, 0, _i)))) { \
+		bc_num_arg_error = true;\
+		break; \
+	}
+
+#define BCMATH_PARSE_PARAMETERS_END() \
+			ZEND_ASSERT(_i == _max_num_args || _max_num_args == (uint32_t) -1); \
+		} while (0); \
+		if (UNEXPECTED(bc_num_arg_error)) { \
+			if (!(_flags & ZEND_PARSE_PARAMS_QUIET)) { \
+				if (EG(exception)) { \
+					return; \
+				} \
+				zend_argument_type_error(_i, "must be of type int, string, or %s, %s given", \
+					ZSTR_VAL(bcmath_number_ce->name), zend_zval_value_name(_arg)); \
+			} \
+			return; \
+		} else if (UNEXPECTED(_error_code != ZPP_ERROR_OK)) { \
+			if (!(_flags & ZEND_PARSE_PARAMS_QUIET)) { \
+				zend_wrong_parameter_error(_error_code, _i, _error, _expected_type, _arg); \
+			} \
+			return; \
+		} \
+	} while (0)
+
+static zend_always_inline zend_result bc_num_from_obj_or_str_or_long_with_err(
+	bc_num *num, size_t *scale, zend_object *obj, zend_string *str, zend_long lval, uint32_t arg_num)
+{
+	if (UNEXPECTED(bc_num_from_obj_or_str_or_long(num, scale, obj, str, lval) == FAILURE)) {
+		zend_argument_value_error(arg_num, "is not well-formed");
+		return FAILURE;
+	}
+	return SUCCESS;
+}
+
+static zend_always_inline zend_result bcmath_check_scale(zend_long scale, bool scale_is_null, uint32_t arg_num)
+{
+	if (UNEXPECTED(!scale_is_null && (scale < 0 || scale > INT_MAX))) {
+		zend_argument_value_error(arg_num, "must be between 0 and %d", INT_MAX);
+		return FAILURE;
+	}
+	return SUCCESS;
+}
+
+PHP_METHOD(BcMath_Number, __construct)
+{
+	zend_string *str = NULL;
+	zend_long lval = 0;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_STR_OR_LONG(str, lval);
+	ZEND_PARSE_PARAMETERS_END();
+
+	bc_num num = NULL;
+	size_t scale;
+	if (bc_num_from_obj_or_str_or_long_with_err(&num, &scale, NULL, str, lval, 1) == FAILURE) {
+		bc_free_num(&num);
+		RETURN_THROWS();
+	}
+
+	bcmath_number_obj_t *intern = get_bcmath_number_from_zval(ZEND_THIS);
```

Comment:
Fixed in https://github.com/php/php-src/pull/13741/commits/24b9ef26753567bd6c959fda0922fa47859c0d1b
Since it is readonly, I made it an error. This behavior is the same as running construct multiple times on a readonly class.